src/modules/Proxy.ycp

```diff
@@ -99,6 +99,68 @@ global define boolean Read() {
     return true;
 }
 
+void WriteSysconfig() {
+    SCR::Write(.sysconfig.proxy.PROXY_ENABLED, enabled?"yes":"no");
+    SCR::Write(.sysconfig.proxy.HTTP_PROXY, http);
+    SCR::Write(.sysconfig.proxy.HTTPS_PROXY, https);
+    SCR::Write(.sysconfig.proxy.FTP_PROXY, ftp);
+    SCR::Write(.sysconfig.proxy.NO_PROXY, no);
+    SCR::Write(.sysconfig.proxy, nil);
+}
+
+/*
+ * Escape backslash characters in .curlrc (bnc#331038)
+ * see also http://curl.haxx.se/docs/manpage.html#-K for escaping rules
+ */
+string EscapeForCurlrc(string s) {
+    if (s == nil) {
+        return nil;
+    }
+    return mergestring (splitstring(s, "\\"), "\\\\");
+}
+
+void WriteCurlrc() {
+    string proxyuser = nil;
+    if (user != "") {
+        proxyuser = user;
+        if (pass != "") proxyuser = user + ":" + pass;
+    }
+
+    map<string, string> options = $[
+        "--proxy-user": proxyuser,
+        // bnc#305163
+        "--proxy":      http,
+        // bsc#923788
+        "--noproxy":    no
```

Comment:
not done in this pr, but variable with name `no` with file scope is quite horrible